src/main/java/org/broadinstitute/hellbender/tools/spark/sv/FindBreakpointEvidenceSpark.java

```diff
@@ -0,0 +1,1160 @@
+                for (final Interval interval : intervals) {
+                    final String seqName = contigs.get(interval.getContig()).getSequenceName();
+                    writer.write(seqName + " " + interval.getStart() + " " + interval.getEnd() + "\n");
+                }
+            } catch (final IOException ioe) {
+                throw new GATKException("Can't write intervals file " + locations.intervalFile, ioe);
+            }
+        }
+
+        return intervals;
+    }
+
+    /** gather some interesting factoids about the reads in aggregate */
+    @VisibleForTesting static ReadMetadata getMetadata( final SAMFileHeader header,
+                                             final int meanBasesPerTemplate ) {
+        final List<SAMReadGroupRecord> groups = header.getReadGroups();
+        final int nGroups = groups.size();
+        final ReadMetadata.ReadGroupFragmentStatistics groupStats =
+                //TODO: get real data
+                new ReadMetadata.ReadGroupFragmentStatistics(400.f, 75.f);
+        final List<ReadMetadata.ReadGroupFragmentStatistics> stats = new ArrayList<>(nGroups);
+        for ( int idx = 0; idx != nGroups; ++idx ) {
+            stats.add(groupStats);
+        }
+        return new ReadMetadata(header, stats, groupStats, meanBasesPerTemplate);
+    }
+
+    /** mean number of sequenced bases per template (i.e., read length if unpaired, twice the read length if paired) */
+    @VisibleForTesting static int getMeanBasesPerTemplate(final JavaRDD<GATKRead> reads ) {
+        return reads
+                .map(ReadCountAndLength::new)
+                .reduce(ReadCountAndLength::new)
+                .getMeanLength();
+    }
+
+    private static void log( final String message ) {
+        System.out.println(dateFormatter.format(System.currentTimeMillis())+message);
+    }
+
+    @VisibleForTesting static class Locations {
+        public final String evidenceDir;
+        public final String intervalFile;
+        public final String qNamesMappedFile;
+        public final String kmerFile;
+        public final String qNamesAssemblyFile;
+
+        public Locations( final String evidenceDir, final String intervalFile,
+                          final String qNamesMappedFile, final String kmerFile, final String qNamesAssemblyFile ) {
+            this.evidenceDir = evidenceDir;
+            this.intervalFile = intervalFile;
+            this.qNamesMappedFile = qNamesMappedFile;
+            this.kmerFile = kmerFile;
+            this.qNamesAssemblyFile = qNamesAssemblyFile;
+        }
+    }
+
+    /** helper class for calculating mean template length */
+    @DefaultSerializer(ReadCountAndLength.Serializer.class)
+    private static final class ReadCountAndLength {
+        private final long count;
+        private final long length;
+
+        ReadCountAndLength( final GATKRead read ) {
+            this.count = 1;
+            this.length = (read.isPaired() ? 2 : 1)*read.getLength();
```

Comment:
Won't both reads in the pair get a ReadCountAndLength fed into them, so aren't you double counting if the reads are paired?